projects/npm-tools/packages/npm-scripts/src/utils/createFederationConfig.js

```diff
@@ -21,9 +21,24 @@ const writeWebpackFederationEntryPoint = require('./writeWebpackFederationEntryP
  * the relation is not used for anything. However, it is better to do it this
  * way, because if we mix all together we end up with a very entangled
  * configuration.
+ *
+ * Note also that when you want to include a project in the federated build to
+ * consume other packages (even if it doesn't publish any package) you must
+ * declare it here.
```

Comment:
When you say "here" does it mean we can declare it dynamically from somewhere, or here here?

As in... `layout-taglib` uses `frontend-js-web` we need to ship a new `npm-scripts` version?

Reply:
Yes, for now we need to release `npm-scripts` versions every time we change this, which is bad. And a source of problems as we have seen with previous PRs, because we need to release `npm-scripts` before we merge the changes in `liferay-portal`.

However I think (but we'll see) that we won't have the same problem again because adding more things to this object doesn't trigger any effect until we activate the federation in `liferay-portal`'s projects (at least that's the theory).

We could argue that this should be extracted ASAP, but on the other hand, we had the preset for `liferay-npm-bundler` which had the "same" problem (we needed to release a new version of it, and then a new version of npm-scripts; much worse, in fact) and nothing happened so, until we advance more in the epic and make this clearer, I think it's OK to leave it here (unless we see a big problem with it in the future).

Reply:
Think of this object as the `imports` section of the default preset we use for the bundler. In fact it's that, because it serves the very same purpose.

Reply:
> We could argue that this should be extracted ASAP

I think it's fine for this to be "in the wrong place", awkward, or inconvenient to update during this phase of development. The design may still change and the most important thing now is proving the viability of webpack federation by rolling it out more broadly. Once it's proven, we can worry about the ergonomics and the update flow.